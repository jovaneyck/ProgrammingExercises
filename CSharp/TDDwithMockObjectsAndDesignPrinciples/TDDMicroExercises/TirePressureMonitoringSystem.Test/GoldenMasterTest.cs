using System;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace TDDMicroExercises.TirePressureMonitoringSystem.Test
{
    [TestFixture]
    public class GoldenMasterTest
    {
        [Test]
        public void NUnitSanityCheck()
        {
            Assert.IsTrue(true);
        }

        [Test]
        [UseReporter(typeof(NUnitReporter))]
        public void AlarmUsedInRealisticScenario()
        {
            //Note that this is a VERY high-level regression test: 
            // the dummy sensor always returns the same value, so the alarm will always be on after the first pressure check.
            StringBuilder s = new StringBuilder();
            for(int i = 0; i < 100; i++)
            {
                Alarm a = new Alarm();
                for(int j = 0; j < 10; j++)
                {
                    s.AppendLine(String.Format("Run {0} {1}: {2}", i,j, a.AlarmOn));
                    a.Check();
                }
            }

            Approvals.Verify(s);
        }
    }
}
