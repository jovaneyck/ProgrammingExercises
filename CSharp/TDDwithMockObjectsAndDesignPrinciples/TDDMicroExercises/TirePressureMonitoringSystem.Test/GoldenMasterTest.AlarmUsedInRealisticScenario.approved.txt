Psi 17: False
Psi 18: False
Psi 19: False
Psi 20: False
Psi 21: False
Psi 22: False
Psi 23: True
Psi 24: True
